flytekit/core/resources.py

```diff
@@ -96,3 +100,48 @@ def convert_resources_to_resource_model(
     if limits is not None:
         limit_entries = _convert_resources_to_resource_entries(limits)
     return task_models.Resources(requests=request_entries, limits=limit_entries)
+
+
+def construct_k8s_pod_spec_from_resources(
+    k8s_pod_name: str,
+    requests: Optional[Resources],
+    limits: Optional[Resources],
+) -> dict[str, Any]:
+    def construct_k8s_pods_resources(resources: Optional[Resources]):
+        if resources is None:
+            return None
+
+        resources_map = {
+            "cpu": "cpu",
+            "mem": "memory",
+            "gpu": "nvidia.com/gpu",
```

Comment:
can you expose this as a parameter? Set its default value to `"nvidia.com/gpu"`

Reply:
Just the gpu value of resource_map?